lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioCoordinatorCacheControl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import io.trino.filesystem.cache.AllowFilesystemCacheOnCoordinator;
+
+import static java.util.Objects.requireNonNull;
+
+public final class AlluxioCoordinatorCacheControl
+        implements AlluxioCacheControl
+{
+    private final boolean enabled;
+
+    // @AllowFilesystemCacheOnCoordinator is an information provided by the connector using file system cache whether it is safe to actually cache on coordinator
+    public AlluxioCoordinatorCacheControl(@AllowFilesystemCacheOnCoordinator boolean allowOnCoordinator, AlluxioFileSystemCacheConfig config)
+    {
+        this.enabled = allowOnCoordinator && requireNonNull(config, "config is null").isCacheOnCoordinator();
```

Comment:
Should we check that `allowOnCoordinator` is not false while `isCacheOnCoordinator()` is true? Hmm, though it might make the defaults a bit tricky.

Reply:
No. These are two separate informations:

config -> user intent to enable caching
annotation -> connector allows caching on coordinator

Reply:
@jkylling we would like to cache metadata on coordinator for supported connectors but disable caching on worker while retaining consistent config for every node